client/signup/steps/design-picker/preview-toolbar.jsx

```diff
@@ -15,18 +15,37 @@ const DesignPickerPreviewToolbar = ( {
 	showDeviceSwitcher,
 	setDeviceViewport,
 	translate,
+	showSiteAddressBar,
+	filterDevicesToShow,
 } ) => {
 	const devices = React.useRef( {
 		computer: { title: translate( 'Desktop' ), icon: computer, iconSize: 36 },
 		tablet: { title: translate( 'Tablet' ), icon: tablet, iconSize: 24 },
 		phone: { title: translate( 'Phone' ), icon: phone, iconSize: 24 },
 	} );
 
+	function filterDevices( selectedDevices ) {
+		// If invalid input, display all possible devices
+		// If valid input, Will filter out any devices that do not exist in the 'possibleDevices' array above
+		if ( Array.isArray( selectedDevices ) && selectedDevices.length > 0 ) {
```

Comment:
What if the consumer passes in [`wrongDevice1`, `wrongDevice2`]? We could probably return `possibleDevices` if `filteredPossibleDevices.length === 0`

Reply:
I updated the `filterDevices` function logic . Let me know if it works and makes sense.